Makefile

```diff
@@ -31,67 +31,50 @@ CONCORD_BFT_CONTAINER_SHELL?=/bin/bash
 CONCORD_BFT_CPPCHECK_PATH?=${CONCORD_BFT_TARGET_SOURCE_PATH}/scripts/run-cppcheck.sh
 CONCORD_BFT_RUN_SIMPLE_TEST?=./build/tests/simpleTest/scripts/testReplicasAndClient.sh
 
 # UDP | TLS | TCP
 CONCORD_BFT_CMAKE_TRANSPORT?=TLS
 ifeq (${CONCORD_BFT_CMAKE_TRANSPORT},TLS)
-   TLS_ENABLED__:=TRUE
-   TCP_ENABLED__:=FALSE
+   TLS_ENABLED__:=ON
+   TCP_ENABLED__:=OFF
 else ifeq (${CONCORD_BFT_CMAKE_TRANSPORT},TCP)
-   TLS_ENABLED__:=FALSE
-   TCP_ENABLED__:=TRUE
+   TLS_ENABLED__:=OFF
+   TCP_ENABLED__:=ON
 else
-   TLS_ENABLED__:=FALSE
-   TCP_ENABLED__:=FALSE
+   TLS_ENABLED__:=OFF
+   TCP_ENABLED__:=OFF
 endif
 
-CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE?='-O3 -g'
-CONCORD_BFT_CMAKE_USE_LOG4CPP?=TRUE
-CONCORD_BFT_CMAKE_BUILD_UTT?=TRUE
-CONCORD_BFT_CMAKE_BUILD_ROCKSDB_STORAGE?=TRUE
-CONCORD_BFT_CMAKE_USE_S3_OBJECT_STORE?=TRUE
-CONCORD_BFT_CMAKE_USE_OPENTRACING?=TRUE
-CONCORD_BFT_CMAKE_USE_PROMETHEUS?=TRUE
-CONCORD_BFT_CMAKE_USE_JAEGER?=TRUE
-CONCORD_BFT_CMAKE_USE_JSON?=TRUE
-CONCORD_BFT_CMAKE_USE_HTTPLIB?=TRUE
-CONCORD_BFT_CMAKE_EXPORT_COMPILE_COMMANDS?=TRUE
-CONCORD_BFT_CMAKE_OMIT_TEST_OUTPUT?=FALSE
-CONCORD_BFT_CMAKE_KEEP_APOLLO_LOGS?=TRUE
-CONCORD_BFT_CMAKE_RUN_APOLLO_TESTS?=TRUE
-CONCORD_BFT_CMAKE_TRANSACTION_SIGNING_ENABLED?=TRUE
-CONCORD_BFT_CMAKE_BUILD_SLOWDOWN?=FALSE
-# Only useful with CONCORD_BFT_CMAKE_BUILD_TYPE:=Release
-CONCORD_BFT_CMAKE_BUILD_KVBC_BENCH?=TRUE
-# Only usefull with CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE=-O0 -g
-CONCORD_BFT_CMAKE_ASAN?=FALSE
-CONCORD_BFT_CMAKE_TSAN?=FALSE
-CONCORD_BFT_CMAKE_UBSAN?=FALSE
-CONCORD_BFT_CMAKE_HEAPTRACK?=FALSE
-CONCORD_BFT_CMAKE_CODECOVERAGE?=FALSE
-CONCORD_BFT_CMAKE_CCACHE?=TRUE
-CONCORD_BFT_CMAKE_USE_FAKE_CLOCK_IN_TIME_SERVICE?=FALSE
-ENABLE_RESTART_RECOVERY_TESTS?=FALSE
-CONCORD_ENABLE_ALL_METRICS?=FALSE
+CONCORD_BFT_CMAKE_BUILD_UTT?=ON
+CONCORD_BFT_CMAKE_OMIT_TEST_OUTPUT?=OFF
+CONCORD_BFT_CMAKE_KEEP_APOLLO_LOGS?=ON
+CONCORD_BFT_CMAKE_RUN_APOLLO_TESTS?=ON
+CONCORD_BFT_CMAKE_ASAN?=OFF
+CONCORD_BFT_CMAKE_TSAN?=OFF
+CONCORD_BFT_CMAKE_UBSAN?=OFF
+CONCORD_BFT_CMAKE_HEAPTRACK?=OFF
+CONCORD_BFT_CMAKE_CODECOVERAGE?=OFF
+CONCORD_BFT_CMAKE_CCACHE?=ON
+ENABLE_RESTART_RECOVERY_TESTS?=OFF
 
 # Our CMake logic won't allow more one of these flags to be raised, so having this if/else logic makes sense
-ifeq (${CONCORD_BFT_CMAKE_ASAN},TRUE)
-	CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE='-O0 -g'
-else ifeq (${CONCORD_BFT_CMAKE_TSAN},TRUE)
-	CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE='-O0 -g'
-else ifeq (${CONCORD_BFT_CMAKE_UBSAN},TRUE)
-	CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE='-O0 -g'
-else ifeq (${CONCORD_BFT_CMAKE_CODECOVERAGE},TRUE)
-	CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE='-O0 -g'
-else ifeq (${CONCORD_BFT_CMAKE_HEAPTRACK},TRUE)
-	CONCORD_BFT_CMAKE_CXX_FLAGS_RELEASE='-O0 -g'
-	ifneq (${RUN_WITH_DEBUG_IMAGE},TRUE)
-    $(info CONCORD_BFT_CMAKE_HEAPTRACK=TRUE: setting RUN_WITH_DEBUG_IMAGE=TRUE to support heaptrack run)
-    RUN_WITH_DEBUG_IMAGE=TRUE
+ifeq (${CONCORD_BFT_CMAKE_ASAN},ON)
+	CONCORD_BFT_CMAKE_BUILD_TYPE=Debug
```

Comment:
Not sure that you'd only want to test leaks in debug settings.
Since release builds are optimized differently, they may introduce leaks which differ from one in debug builds.

Reply:
I haven't changed this behavior.
We are going to revise the whole sanitizer thing with gcc